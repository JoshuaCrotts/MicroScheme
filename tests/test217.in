;
; Problems 2&3 A6 B521
;

(define (zero? x) (= x 0))

(define star
  (lambda (m)
    (lambda (n)
      (* m n))))

((star 2) 3)
((star ((star 2) 3)) 5)

(define times
  (lambda (ls)
    (cond
      [(null? ls) 1]
      [(zero? (car ls)) 0]
      [else (* (car ls) (times (cdr ls)))])))
(times '(1 2 3 4 5))
(times '(1 2 3 0 3))
