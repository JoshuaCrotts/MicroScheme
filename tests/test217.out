6
30
120
0
